assets/app/view/game/corporate_buy_shares.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require 'view/game/actionable'
+
+module View
+  module Game
+    class CorporateBuyShares < Snabberb::Component
+      include Actionable
+
+      needs :game, store: true
+
+      def render
+        @step = @game.round.active_step
+        @current_actions = @step.current_actions
+        @entity ||= @game.current_entity
+        children = []
+
+        children.concat(render_sources) if @step.current_actions.include?('corporate_buy_shares')
+
+        h('div.margined', children.compact)
+      end
+
+      def render_sources
+        props = {
+          style: {
+            display: 'inline-block',
+            verticalAlign: 'top',
+          },
+        }
+
+        @step.source_list(@entity).map do |source|
+          children = []
+          if source.player?
+            children << h(Player, player: source, game: @game)
+            children << render_player_input(source)
+          elsif source.corporation? && @game.corporation_available?(source)
+            children << h(Corporation, corporation: source)
+            children << render_corporation_input(source)
+          end
+          h(:div, props, children)
+        end.compact
```

Comment:
why do you call compact here

Reply:
I started with block 73-85 from view/game/round/stock.rb but I see .compact is used there because the map can produce nils. However, this lead me to finding a tiny bug with my code where an empty div was produced if the corporation wasn't available. I will move that check into a "next" line and now the .compact should be applicable.

If it is still not applicable, then I'm not sure why we do it in stock.rb either?

Reply:
it's applicable in view/lround/stock because it uses next which causes nils